### PR TITLE
Sets up the initial project structure

### PR DESCRIPTION
**Pull request by duncgibbs**
I would take a look at `tsconfig.json`. I set some of the typical features, but the documentation is pretty good if you want to include anything else.

### CODE REVIEW

**Comment by duncgibbs** [comment]
I didn't set up linting. That may or may not be something you want. If so, I recommend `eslint`.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "test": "jest",
+    "build": "webpack-cli",
+    "watch": "webpack-cli -w"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/tylercecil/chomTree.git"
+  },
+  "author": "Tyler Cecil",
+  "license": "ISC",
```
Comment (tylercecil) [inline package.json:15]:
```suggestion
  "license": "MIT",
```

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+console.log('Duncan rules. Tyler drools.');
```
Comment (tylercecil) [inline src/index.ts:1]: Acceptable.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
```
Comment (tylercecil) [inline tsconfig.json:5]: Target is what we are building into, yeah? Any reason what ES5 over the default?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "chomtree",
```
Comment (tylercecil) [inline package.json:2]:
```suggestion
  "name": "chomTree",
```

Comment (tylercecil) [reply package.json:2]: Follow up — do changes to package.json require package-lock.json to be regenerated?

File: package.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  "scripts": {
+    "test": "jest",
+    "build": "webpack-cli",
+    "watch": "webpack-cli -w"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/tylercecil/chomTree.git"
+  },
+  "author": "Tyler Cecil",
```
Comment (tylercecil) [inline package.json:14]:
```suggestion
  "author": "Tyler Cecil <tyler.i.cecil@gmail.com> (https://github.com/tylercecil)",
  "contributors": [
    "Tyler Cecil <tyler.i.cecil@gmail.com> (https://github.com/tylercecil)",
    "Duncan Gibbs <duncgibbs@gmail.com> (https://github.com/duncgibbs)"
  ],
```

File: tsconfig.json (hunk 3 above)
Comment (duncgibbs) [reply tsconfig.json:5]: So many reasons. Actually, we may want to go with ES6 if you care about immutability. (Every browser on Earth supports both ES5 and ES6.)

File: package.json (hunk 4 above)
Comment (duncgibbs) [reply package.json:2]: `package-lock.json` is regenerated whenever you run `npm install` or `npm update`.

Comment (duncgibbs) [reply package.json:2]: Also, only lowercase letters are allowed.

File: tsconfig.json (hunk 3 above)
Comment (tylercecil) [reply tsconfig.json:5]: I expected ES6. But actually, does it matter for the target language? Anyway, I'm fine with es6.

File: package.json (hunk 4 above)
Comment (tylercecil) [reply package.json:2]: :+1:

**Comment by tylercecil** [comment]
Go ahead and merge master, so we can get a CI build. Then LGTM.

**Pull request merged**